bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/IndexPersistenceMgr.java

```diff
@@ -494,7 +494,20 @@ int getOpenFileLimit() {
     private void relocateIndexFileAndFlushHeader(long ledger, FileInfo fi) throws IOException {
         File currentDir = getLedgerDirForLedger(fi);
         if (ledgerDirsManager.isDirFull(currentDir)) {
-            moveLedgerIndexFile(ledger, fi);
+            try {
+                moveLedgerIndexFile(ledger, fi);
+            } catch (NoWritableLedgerDirException nwe) {
+                /*
+                 * if there is no other indexDir, which could accommodate new
+                 * indexFile but the current indexDir has enough space
+                 * (minUsableSizeForIndexFileCreation) for this flushHeader
+                 * operation, then it is ok to proceed without moving
+                 * LedgerIndexFile.
+                 */
+                if (!ledgerDirsManager.isDirWritableForNewIndexFile(currentDir)) {
```

Comment:
Add info log message here.

Reply:
did go through the calls, "ledgerDirsManager.isDirWritableForNewIndexFile" call would log in the case of true return value and in the case of false it would throw exception, which would get logged eventually. So log line is not necessary here.